examples/tv-casting-app/android/App/app/src/main/jni/com/matter/casting/support/IdentificationDeclarationOptions.java

```diff
@@ -0,0 +1,108 @@
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.matter.casting.support;
+
+import android.util.Log;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class contains the optional parameters used in the IdentificationDeclaration Message, sent
+ * by the Commissionee to the Commissioner. The options specify information relating to the
+ * requested UDC commissioning session.
+ */
+public class IdentificationDeclarationOptions {
+  static final String TAG = IdentificationDeclarationOptions.class.getSimpleName();
+  private static final int CHIP_DEVICE_CONFIG_UDC_MAX_TARGET_APPS = 10;
+
+  public IdentificationDeclarationOptions() {}
+
+  /**
+   * Feature: Target Content Application Flag to instruct the Commissioner not to display a Passcode
+   * input dialog, and instead send a CommissionerDeclaration message if a commissioning Passcode is
+   * needed.
+   */
+  public boolean mNoPasscode = false;
+  /**
+   * Feature: Coordinate Passcode Dialogs Flag to instruct the Commissioner to send a
+   * CommissionerDeclaration message when the Passcode input dialog on the Commissioner has been
+   * shown to the user.
+   */
+  public boolean mCdUponPasscodeDialog = false;
+  /**
+   * Feature: Commissioner-Generated Passcode Flag to instruct the Commissioner to use the
+   * Commissioner-generated Passcode for commissioning.
+   */
+  public boolean mCommissionerPasscode = false;
+  /**
+   * Feature: Commissioner-Generated Passcode Flag to indicate whether or not the Commissionee has
+   * obtained the Commissioner Passcode from the user and is therefore ready for commissioning.
+   */
+  public boolean mCommissionerPasscodeReady = false;
+  /**
+   * Feature: Coordinate Passcode Dialogs Flag to indicate when the Commissionee user has decided to
+   * exit the commissioning process.
+   */
+  public boolean mCancelPasscode = false;
+  /**
+   * Feature: Target Content Application The set of content app Vendor IDs (and optionally, Product
+   * IDs) that can be used for authentication. Also, if TargetAppInfo is passed in,
+   * VerifyOrEstablishConnection() will force User Directed Commissioning, in case the desired
+   * TargetApp is not found in the on-device CastingStore.
+   */
+  private List<TargetAppInfo> mTargetAppInfos = new ArrayList<>();
+
+  public boolean addTargetAppInfo(TargetAppInfo targetAppInfo) {
+    Log.d(TAG, "addTargetAppInfo()");
+    if (mTargetAppInfos.size() >= CHIP_DEVICE_CONFIG_UDC_MAX_TARGET_APPS) {
+      Log.e(
+          TAG,
+          "addTargetAppInfo() failed to add TargetAppInfo, max list size is {0}"
+              + CHIP_DEVICE_CONFIG_UDC_MAX_TARGET_APPS);
+      return false;
+    }
+    mTargetAppInfos.add(targetAppInfo);
+    return true;
+  }
+
+  public List<TargetAppInfo> getTargetAppInfoList() {
+    return mTargetAppInfos;
+  }
+
+  public void logDetail() {
```

Comment:
Override the Java toString() function instead of rolling a new logDetail function for this.